algorithm.go

```diff
@@ -0,0 +1,87 @@
+package medley
+
+import (
+	"hash"
+	"hash/fnv"
+	"strings"
+
+	"github.com/spaolacci/murmur3"
+)
+
+const (
+	// AlgorithmFNV is the configuration value for an fnv.New64a algorithm
+	AlgorithmFNV = "fnv"
+
+	// AlgorithmMurmur3 is the configuration value for a murmur3.New64 algorithm
+	AlgorithmMurmur3 = "murmur3"
+)
+
+// UnknownAlgorithmError indicates that no algorithm could be created using
+// the given Name
+type UnknownAlgorithmError struct {
+	// Name is the name which is unrecognized.  This will never be blank,
+	// as a blank name is interpreted as AlgorithmDefault.
+	Name string
+}
+
+// Error fulfills the error interface
+func (e *UnknownAlgorithmError) Error() string {
+	var o strings.Builder
```

Comment:
`fmt.Sprintf` has to parse the format string, do validation, reflect each argument to determine how to print it, etc.  It's generally not a good idea to use in the hot path.

For simple things like this, I personally prefer to use `strings.Builder` just because I don't find it less readable.  If there were more arguments, I would definitely use `fmt.Sprintf`.